src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/Handler/RefundItemsTransactionHandler.php

```diff
@@ -0,0 +1,251 @@
+     */
+    protected function getRefundedUniqueItems(array $itemTransfers): ArrayObject
+    {
+        $uniqueItemTransfers = [];
+        $uniqueItemIds = [];
+
+        foreach ($itemTransfers as $itemTransfer) {
+            if (!in_array($itemTransfer->getIdSalesOrderItem(), $uniqueItemIds)) {
+                $uniqueItemTransfers[] = $itemTransfer;
+                $uniqueItemIds[] = $itemTransfer->getIdSalesOrderItem();
+            }
+        }
+
+        $uniqueItemTransfers = new ArrayObject($uniqueItemTransfers);
+
+        return $uniqueItemTransfers;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\TransactionMetaTransfer $transactionMetaTransfer
+     *
+     * @return array
+     */
+    protected function getOrderItemsGroupedByTransaction(TransactionMetaTransfer $transactionMetaTransfer): array
+    {
+        $itemsByTransactions = [];
+
+        $refundedItemMap = $this->mapItemsAmount($transactionMetaTransfer->getItems());
+
+        $paymentBraintreeOrderItemTransfers = $this->braintreeRepository
+            ->findPaymentBraintreeOrderItemsByIdsSalesOrderItem(array_keys($refundedItemMap));
+
+        foreach ($paymentBraintreeOrderItemTransfers as $paymentBraintreeOrderItemTransfer) {
+            $paymentBraintreeTransactionStatusLogTransfer = $this->braintreeRepository
+                ->findPaymentBraintreeTransactionStatusLogQueryByPaymentBraintreeOrderItem($paymentBraintreeOrderItemTransfer->getIdPaymentBraintreeOrderItem());
+
+            $amount = $itemsByTransactions[$paymentBraintreeTransactionStatusLogTransfer->getTransactionId()][static::KEY_AMOUNT] ?? 0;
+
+            $itemsByTransactions[$paymentBraintreeTransactionStatusLogTransfer->getTransactionId()] = [
+                static::KEY_PAYMENT_ID => $paymentBraintreeOrderItemTransfer->getFkPaymentBraintree(),
+                static::KEY_AMOUNT => $amount + $refundedItemMap[$paymentBraintreeOrderItemTransfer->getFkSalesOrderItem()],
+            ];
+        }
+
+        return $itemsByTransactions;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
+     *
+     * @return \Generated\Shared\Transfer\RefundTransfer
+     */
+    protected function removeShipmentExpense(RefundTransfer $refundTransfer): RefundTransfer
+    {
+        $expenses = [];
+
+        foreach ($refundTransfer->getExpenses() as $expenseTransfer) {
+            if ($expenseTransfer->getType() !== ShipmentConstants::SHIPMENT_EXPENSE_TYPE) {
+                $expenses[] = $expenseTransfer;
+            }
+        }
+
+        $refundTransfer->setExpenses(new ArrayObject($expenses));
+
+        return $refundTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
+     *
+     * @return \Generated\Shared\Transfer\ExpenseTransfer|null
+     */
+    protected function getShipmentExpenseTransfer(RefundTransfer $refundTransfer): ?ExpenseTransfer
```

Comment:
is it inherited old method? If no - get* methods should always return smth. If method can return null - it should be named with find* prefix